### PR TITLE
CORE-12139 move privacysalt to runtime-os

### PR DESCRIPTION
**Pull request by vlajos**
CORE-12139 move privacysalt to runtime-os

### CODE REVIEW

File: libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/PrivacySalt.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+@CordaSerializable
+@DoNotImplement
+interface PrivacySalt {
+    /**
+     * Gets the value of the privacy salt.
+     *
+     * @return Returns the value of the privacy salt.
+     */
+    fun getBytes(): ByteArray
```
Comment (blsemo) [inline libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/PrivacySalt.kt]: Shouldn't this be a val property?

File: components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerRepositoryTest.kt [hunk 2]
```diff
@@ -135,7 +135,7 @@ class ConsensualLedgerRepositoryTest {
 
             entityFactory.createConsensualTransactionEntity(
                 signedTransaction.id.toString(),
-                signedTransaction.wireTransaction.privacySalt.bytes,
+                signedTransaction.wireTransaction.privacySalt.getBytes(),
```
Comment (blsemo) [inline components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerRepositoryTest.kt]: See comment below about property on the interface

**Review by blsemo — COMMENTED** [review]
I think if you turn `fun getBytes()` into `val bytes` on the interface, half of the changes will disappear.

File: libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/PrivacySalt.java
Comment (blsemo) [inline libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/PrivacySalt.java:1]: That file is now in the wrong location - should be in the java hierarchy, not kotlin

**Review by blsemo — COMMENTED** [review]
Looks good apart from one wrong file location

**Review by blsemo — DISMISSED** [review]
LGTM

**Review by lankydan — APPROVED** [review]

**Pull request merged**